tests/test_preprocess.py

```diff
@@ -0,0 +1,46 @@
+import sys
+import os
+# Set the path to the src directory
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../src'))) 
+from src import preprocess
+import numpy as np
+import pytest
+import json
+
+
+@pytest.mark.fast
+def test_fake_data():
+    sys.argv = ["", "tests/testdata"]
+    preprocess.main()
```

Comment:
Open for discussion: But isn't it better to have fixtures or some other logic for saving/loading or even hardcoding data?

Some online discussions: 
https://stackoverflow.com/questions/34466027/what-is-conftest-py-for-in-pytest
https://stackoverflow.com/questions/54330834/what-is-a-correct-approach-to-manage-test-data-using-pytest

Repeating paths to test data is a pain. We can at least have those hardcoded for pytest in ways that pytest makes possible for us.

Reply:
This seems like a good improvement for later.